aten/src/ATen/native/TensorShape.cpp

```diff
@@ -194,6 +193,17 @@ Tensor & _cat_out_cpu(TensorList tensors, int64_t dim, Tensor& result) {
 
   // skip resizing if size of result is same as expected
   if (result.sizes() != result_size) {
+    // raise a warning while resizing if output has one or more elements
+    if (result.numel() != 0) {
```

Comment:
this might be a good opportunity to switch to structured kernels for `cat`.

cc. @SplitInfinity who might be able to help on the specifics

Reply:
I'm not fully aware of the porting to structured kernels, I'm assuming this is the corresponding [RFC](https://github.com/pytorch/rfcs/blob/rfc-0005/RFC-0005-structured-kernel-definitions.md)?

Shouldn't this be left to another PR, or do you feel it is within the scope of this one?

Reply:
If `cat` is not ported to `structured` in this PR, then use `resize_output` from this file. https://github.com/pytorch/pytorch/blob/c7db642a72d80facf9c8a17f42ac0bb04d04723f/aten/src/ATen/native/Resize.cpp#L38-L39